app/data/views.py

```diff
@@ -315,3 +348,55 @@ def create(self, request, *args, **kwargs):
 
     # TODO: If we switch to a Django/ORM database backend, we can subclass AbortableTask
     # and allow cancellation as well.
+
+
+class JarFileRenderer(renderers.BaseRenderer):
+    """Renderer for downloading JARs.
+
+    http://www.django-rest-framework.org/api-guide/renderers/#data
+    """
+    media_type = 'application/java-archive'
+    charset = None
+    render_style = 'binary'
+    format = 'jar'
+
+    def render(self, data, media_type=None, renderer_context=None, format=None):
+        return data
+
+
+class AndroidSchemaModelsViewSet(viewsets.ViewSet):
+    """A view for interacting with Android jar build jobs."""
+
+    permissions_classes = (IsAuthenticated,)
+    renderer_classes = [JarFileRenderer] + api_settings.DEFAULT_RENDERER_CLASSES
+
+    def finalize_response(self, request, response, *args, **kwargs):
+        response = super(AndroidSchemaModelsViewSet, self).finalize_response(request, response, *args, **kwargs)
+        if isinstance(response.accepted_renderer, JarFileRenderer):
+            response['content-disposition'] = 'attachment; filename=models.jar'
+        return response
+
+    def retrieve(self, request, pk=None):
+        """Return the model jar if found, or kick of a task to create it if not found"""
+
+        uuid = str(pk)
+        if not uuid:
+            return Response({'errors': {'uuid': 'This parameter is required'}},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        redis_conn = get_redis_connection('jars')
+        found_jar = redis_conn.get(uuid)
```

Comment:
It will create another task. Building takes less than ten seconds once the gradle dependencies are cached, so this shouldn't happen much or have much effect.